source/pdo_sqlsrv/pdo_stmt.cpp

```diff
@@ -401,9 +401,12 @@ int pdo_sqlsrv_stmt_close_cursor(pdo_stmt_t *stmt TSRMLS_DC)
 
         // to "close the cursor" means we make the statement ready for execution again.  To do this, we 
         // skip all the result sets on the current statement.
-        while( driver_stmt->past_next_result_end == false ) {
-
-            core_sqlsrv_next_result( driver_stmt TSRMLS_CC );
+        // If the statement has not been executed there are no next results to iterate over.
+        if ( driver_stmt->executed == true )
+        {
+            while( driver_stmt->past_next_result_end == false ) {
+                core_sqlsrv_next_result( driver_stmt TSRMLS_CC );
```

Comment:
Hi @deeky666 the suggested fix by @v-dareck is actually more efficient to fix the problem you originally raised because there's nothing to cancel in your repro scenario. 

Besides, in normal circumstances when the query has actually been executed closeCursor() will do the clean up and finalize output parameters etc.

Please let us know if this have clarified things for you.